src/error.rs

```diff
@@ -71,6 +82,8 @@ pub enum Error {
     /// A standard error with a string description;
     /// a more specific error should generally be used.
     DefaultError(String),
+    #[cfg(feature = "ssl")]
+    SslError(String),
```

Comment:
I'd be okay with that. In their current form, the error messages are admittedly not that helpful.